### ISSUE TITLE
Not able to find fieldset element using `getByRole('group', { name: 'Name'})` 

### ISSUE THREAD

**Issue by willamzv** [testing-library/dom-testing-library#576]
- `@testing-library/dom` version:  "^7.2.2"
- Testing Framework and version:
 - `@testing-library/react` version: "^10.0.4"
  
- DOM Environment: "jest-environment-jsdom-sixteen": "^1.0.3"
  

Node version: 13.7.0

### Relevant code or config:

```js
test("fieldset accessible name", () => {
  const { getByRole } = render(
    <>
      <fieldset>
        <legend>My Legend</legend>
        <label>
          My Input
          <input type="text" />
        </label>
      </fieldset>
    </>
  );

  const fieldset = getByRole("group", { name: /my legend/i });
  expect(within(fieldset).getByLabelText(/my input/i)).not.toBeNull();
});

```

### What you did:

Following the suggested guidance from [Which query should I use FAQ](https://testing-library.com/docs/guide-which-query)
I am trying to get a fieldset element by its role and name.

### What happened:

```
Unable to find an accessible element with the role "group" and name `/fieldset/i`

Here are the accessible roles:

  document:

  Name "":
  <body />

  --------------------------------------------------
  group:

  Name "":
  <fieldset />

  --------------------------------------------------
  textbox:

  Name "My Input":
  <input
    type="text"
  />

  --------------------------------------------------

<body>
  <div>
    <fieldset>
      <legend>
        Legend
      </legend>
      <label>
        My Input
        <input
          type="text"
        />
      </label>
    </fieldset>
  </div>
</body>

```

### Reproduction:

https://codesandbox.io/s/fieldset-accessible-name-69lhg?file=/src/__tests__/App.test.js

### Problem description:

1. Trying to get a fieldset by its role: "group", the accessible name for the "group" is not associated with the legend element.

```
  --------------------------------------------------
  group:

  Name "":
  <fieldset />

  --------------------------------------------------
```

### Suggested solution:

1. As described in [WAI-ARIA Name with legend](https://www.w3.org/TR/wai-aria-practices-1.1/#naming_with_legend):
> The HTML fieldset element can be used to group form controls, and the legend element can be used to give the group a name.

Consider assigning legend element as accessible name for a fieldset.

**Comment by eps1lon** [comment]
The fix is likely similar to #570. Instead of the `<caption>` for a `<table>` we should use the `<legend>` for a `<fieldset>`.

**Comment by MatanBobi** [comment]
@dietergeerts thanks for reaching out :)
It's not that the `name` attribute on a `fieldset` doesn't work anymore, it's that it never did work.
If you look at the [accessible name computation](https://www.w3.org/TR/html-aam-1.0/#fieldset-element-accessible-name-computation) for `fieldset` element, a `name` isn't one of the attributes being used.
You can use `aria-label`, `aria-labelledby`, a `legend` element or a `title` attribute.

![image](https://user-images.githubusercontent.com/12711091/137261843-276d318f-9fa1-453f-b5ef-3676505fa602.png)
